rmm/monitor/src/event/mainloop.rs

```diff
@@ -38,42 +35,48 @@ impl Mainloop {
     }
 
     pub fn boot_complete(&mut self, smc: SecureMonitorCall) {
-        let ctx = Context {
-            cmd: rmi::BOOT_COMPLETE,
-            arg: [rmi::BOOT_SUCCESS, 0, 0, 0],
-            ..Default::default()
-        };
+        let mut ctx = Context::new(rmi::BOOT_COMPLETE);
+        ctx.init_arg(&[rmi::BOOT_SUCCESS]);
+
         self.add_event_handlers();
         self.dispatch(smc, ctx);
     }
 
     pub fn dispatch(&self, smc: SecureMonitorCall, ctx: Context) {
-        let ret = smc.call(ctx.cmd, ctx.arg);
-        let ctx = Context {
-            cmd: ret[0],
-            arg: [ret[1], ret[2], ret[3], ret[4]],
-            ..Default::default()
-        };
-        self.tx.send(ctx);
+        let ret = smc.call(ctx.cmd(), ctx.arg_slice());
+        let cmd = ret[0];
+
+        rmi::constraint::validate(
+            cmd,
+            |arg_num, ret_num| {
+                let mut ctx = Context::new(cmd);
+                ctx.init_arg(&ret[1..arg_num]);
```

Comment:
I think this should be fixed. As you know, it is half open range.  

```suggestion
                ctx.init_arg(&ret[1.. 1+ arg_num]);
```

Anyway, how does it passed..?

Reply:
Correct, it works that way.
If arg_num is 3,
ret[0] is used to indicate "command", ret[1] and ret[2] are arguments.
arg[0] = ret[1];
arg[1] = ret[2];
So, ret[1..arg_num] == ret[1..3] --> this is a correct behavior.